cli/src/main/java/com/devonfw/tools/ide/git/GitContextImpl.java

```diff
@@ -300,9 +298,8 @@ public Path findGit() {
         return null;
       }
     } else {
-      Path binaryGitPath = this.context.getPath().findBinary(gitPath);
       if (gitPath == binaryGitPath) {
```

Comment:
This has to go before `if (SystemInfoImpl.INSTANCE.isWindows()) {`.
So only if we did not find bash in our PATH, we can do the Windows workaround.
If it helps you again extract private method `findGitOnPath()` and `findGitOnWindowsViaBash()`.